### PR TITLE
Parser for custom Markdown

### PR DESCRIPTION
**Pull request by nirmal-suthar**
This is a simple parser capable of parsing:
- description
- metadata
- data properties (with 2 level nested list)

### CODE REVIEW

**Review by zvr — COMMENTED** [review]
Two general comments (not per line):

You are mixing lexical and syntactic parsing, which may result into problems ahead. Try to keep these separate.
To give an example, "description_header: newlines TWO_HASHES DESCRIPTION" should be handled in the parsing stage, not the lexical one.

Since you're parsing, for example, a class, it would be better to have the result as a dict, instead of a nested list with 'what_this_is_about' strings on each expression. Any further processing will be much easier than traversing simple lists.

**Comment by nirmal-suthar** [comment]
> You are mixing lexical and syntactic parsing, which may result into problems ahead. Try to keep these separate.
> To give an example, "description_header: newlines TWO_HASHES DESCRIPTION" should be handled in the parsing stage, not the lexical one.

I have separated the lexical and syntactic parsing in the latest commit. The only concern is that this create shift/reduce conflicts which I have tried to fix by merging hashes and description together.
 
> Since you're parsing, for example, a class, it would be better to have the result as a dict, instead of a nested list with 'what_this_is_about' strings on each expression. Any further processing will be much easier than traversing simple lists.

I was only experimenting with the parser so used list, in latest commit I have attempted to implement more practical structure.

File: src/driver.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+from argparse import ArgumentParser
+from spec_parser import SpecParser
+
+
+def get_args():
+
+    argparser = ArgumentParser(prog='spec-parser',
+                               description='SPDX specification parser')
+
+    argparser.add_argument('model', type=str,
```
Comment (zvr) [inline src/driver.py:11]: One argument, the top-level directory.

**Review by zvr — CHANGES_REQUESTED** [review]

File: src/driver.py (hunk 1 above)
Comment (nirmal-suthar) [reply src/driver.py]: Resolved! now it expects directory `spec-dir` containing all namespaces..

**Pull request merged**